fury/ui/elements.py

```diff
@@ -3084,6 +3084,113 @@ def directory_click_callback(self, i_ren, _obj, listboxitem):
         i_ren.event.abort()
 
 
+class DrawShapeGroup:
+    def __init__(self):
+        self.grouped_shapes = []
+        self._scene = None
+
+        # Group rotation slider
+        self.group_rotation_slider = RingSlider2D(initial_value=0,
+                                                  text_template="{angle:5.1f}°")
+
+        self.group_rotation_slider.set_visibility(False)
+
+        def update_rotation(slider):
+            angle = slider.value
+            previous_angle = slider.previous_value
+            rotation_angle = angle - previous_angle
+
+            for shape in self.grouped_shapes:
+                current_center = shape.center
+                shape.rotate(np.deg2rad(rotation_angle))
+                shape.update_shape_position(current_center - shape.drawpanel.position)
+
+        self.group_rotation_slider.on_change = update_rotation
+
+    def add(self, shape):
+        """Add shape to the group.
+
+        Parameters
+        ----------
+        shape : DrawShape
+
+        """
+        if self.is_present(shape):
+            self.remove(shape)
+        else:
+            if self.is_empty():
+                shape.drawpanel.update_shape_selection(shape)
+                self.add_rotation_slider(self._scene)
+                self.group_rotation_slider.set_visibility(True)
+            self.grouped_shapes.append(shape)
+            shape.is_selected = True
+            shape.rotation_slider.set_visibility(False)
+
+            self.group_rotation_slider.center = shape.rotation_slider.center
+
+    def remove(self, shape):
+        """Remove shape from the group.
+
+        Parameters
+        ----------
+        shape : DrawShape
+
+        """
+        self.grouped_shapes.remove(shape)
+        shape.is_selected = False
+
+    def clear(self):
+        """Remove all the shapes from the group.
+
+        """
+        if self.is_empty():
+            return
+        self._scene.rm(*self.group_rotation_slider.actors)
+        for shape in self.grouped_shapes:
+            shape.is_selected = False
+        self.grouped_shapes = []
+
+    def is_present(self, shape):
+        """Check whether the shape is present in the group.
+
+        Parameters
+        ----------
+        shape : DrawShape
+
+        """
+        if shape in self.grouped_shapes:
+            return True
+        return False
+
+    def is_empty(self):
+        """Return whether the group is empty or not.
+
+        """
+        return not bool(len(self.grouped_shapes))
+
+    def update_position(self, offset):
+        """Update the position of all the shapes in the group.
+
+        Parameters
+        ----------
+        offset : (float, float)
+            Distance by which each shape is to be translated.
+
+        """
+        for shape in self.grouped_shapes:
+            shape.update_shape_position(shape.center + offset)
```

Comment:
In update_position function, keep the distance of the shapes (with each other) fixed. Otherwise, the shapes collapse in one position as you move them near the borders. To do this, the "shape.center" should be replace by "group.center" which represents the center of grouped shapes.